Examples/ObjectiveC/SwitchStylesExample.m

```diff
@@ -0,0 +1,90 @@
+#import "SwitchStylesExample.h"
+@import Mapbox;
+
+NSString *const MBXExampleSwitchStyles = @"SwitchStylesExample";
+
+@interface SwitchStylesExample () <MGLMapViewDelegate, UIPickerViewDataSource>
+
+@property (nonatomic) MGLMapView *mapView;
+@property (nonatomic) UIPickerView *stylePicker;
+@property (nonatomic) UIButton *button;
+@property (nonatomic, readonly) NSArray<NSDictionary *> *styleURLs;
+
+@end
+
+@implementation SwitchStylesExample
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    _styleURLs = @[
+                    @{@"Mapbox Light":[MGLStyle lightStyleURL]},
+                    @{@"Mapbox Streets":[MGLStyle streetsStyleURL]},
+                    @{@"Mapbox Dark":[MGLStyle darkStyleURL]},
+                    @{@"Mapbox Outdoors": [MGLStyle outdoorsStyleURL]},
+                    @{@"Mapbox Satellite": [MGLStyle satelliteStyleURL]}
+                  ];
+
+    self.mapView = [[MGLMapView alloc] initWithFrame: CGRectMake(0, 0, self.view.bounds.size.width, self.view.bounds.size.height - 50)];
```

Comment:
You should investigate extending the map underneath the bottom UI and then setting this as an inset on the map view.